packages/telemetry/src/appInsights.ts

```diff
@@ -0,0 +1,206 @@
+   */
+  public sendTelemetryException(exception: Error, attributes: Attributes = {}): void {
+    this.logger.debug(`Sending telemetry exception: ${exception.message}`);
+    const { properties, measurements } = buildPropertiesAndMeasurements(attributes);
+    this.appInsightsClient.trackException({ exception, properties, measurements });
+  }
+
+  /**
+   * Publishes diagnostic information to app insights dashboard
+   * @param message {string} - trace message to sen to app insights.
+   * @param properties {Properties} - map of properties to publish alongside the event.
+   */
+  public sendTelemetryTrace(traceMessage: string, properties?: Properties): void {
+    this.logger.debug(`Sending telemetry: trace ${traceMessage}`);
+    this.appInsightsClient.trackTrace({ message: traceMessage, properties });
+  }
+
+  /**
+   * Publishes metric to app insights dashboard
+   * @param name {string} - name of the metric you want published
+   * @param value {number} - value of the metric
+   * @param properties {Properties} - map of properties to publish alongside the event.
+   */
+  public sendTelemetryMetric(metricName: string, value: number, properties?: Properties): void {
+    this.logger.debug(`Sending telemetry metric: ${metricName}`);
+    this.appInsightsClient.trackMetric({ name: metricName, value, properties });
+  }
+
+  public start() {
+    // Start data collection services
+    appInsights.start();
+  }
+
+  public stop() {
+    this.appInsightsClient.flush();
+    appInsights.dispose();
+  }
+
+  /**
+   * Initiates the app insights client
+   */
+  private createAppInsightsClient(): void {
+    this.logger.debug('creating appInsightsClient');
+
+    appInsights.setup(this.options.key);
+
+    this.appInsightsClient = appInsights.defaultClient;
+    this.appInsightsClient.commonProperties = this.buildCommonProperties();
+    this.appInsightsClient.context.tags = this.buildContextTags();
+
+    if (isAsimovKey(this.options.key)) {
+      this.appInsightsClient.config.endpointUrl = AppInsights.ASIMOV_ENDPOINT;
+    }
+  }
+
+  /**
+   * Builds the properties to send with every event
+   * @return {Properties} map of base properites and properties provided when class was created
+   */
+  private buildCommonProperties(): Properties {
+    const baseProperties: Properties = {
+      'common.cpus': getCpus(),
+      'common.os': os.platform(),
+      'common.platformversion': getPlatformVersion(),
+      'common.systemmemory': getSystemMemory(),
+      'common.usertype': this.env.getString('SFDX_USER_TYPE') || 'normal'
+    };
+    return Object.assign(baseProperties, this.options.commonProperties);
+  }
+
+  /**
+   * Builds the context tags for appInsightsClient
+   * @return {Properties} map of tags to add to this.appInsightsClient.context.tags
+   */
+  private buildContextTags(): Properties {
+    const currentTags = this.appInsightsClient ? this.appInsightsClient.context.tags : {};
```

Comment:
Either here or somewhere close to where you start the service you should have a `this.appInsightsClient.context.tags['ai.user.id']` in order to make the metrics be per user.

Reply:
Thanks! Added.